### DISCUSSION TITLE
How does istio prevent malicious outbound traffic, if it can be disabled from within the POD

### DISCUSSION THREAD

**Discussion by MartinKosicky** [istio/istio#53325] [Q&A]
Hello. Regarding security over past few years we noticed that there is currently a big vector of attack on the supplied libraries and so on. There can be malicisous code that works like a botnet from inside your pod in some of your 3rd party libraries. These botnest usually need some comand&control center, which can be anything that gives commands to the pods. It can be a queue, someone elses database, or some URL.  To overcome this I thought that we can rely on istio and the external communication policy, to prevent all unwanted communication going outside. But then someone told me that the istio proxy can be turned off from inside the POD using some special URL.   So the question is, if istio at all can provide some security over unwanted outbound communication?

**Comment by Raven0304** [comment]
salam

**Reply by Raven0304** [reply]
salam qaqa

**Comment by howardjohn** [comment]
On its own a sidecar cannot enforce egress policies. However it can route to an egress gateway which does this enforcement, and then network policy can be used on top to ensure all egress traffic actually goes through the gateway. https://istio.io/latest/docs/tasks/traffic-management/egress/egress-gateway/

**Comment by MartinKosicky** [comment]
But its the sidecar that routes traffic to egress gateway

**Reply by howardjohn** [reply]
The sidecar routes traffic to the egress gateway, but the network policy blocks any traffic that does NOT follow this pattern. This way, if the attacker somehow bypasses the routing to egress gw, their traffic is dropped.
